lib/charliecloud.py

```diff
@@ -341,7 +368,17 @@ def layers_read(self):
          except tarfile.TarError as x:
             FATAL("cannot open: %s: %s" % (path, x))
          members = collections.OrderedDict([(m, None) for m in members_list])
-         layers[lh] = TT(fp, members)
+         if (lh in layers and members):
```

Comment:
I think the precedence here is `(lh in layers) and members`, not `lh in (layers and members)` which it seems this is looking for.